src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/LocalAppContextSwitches.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class LocalAppContextSwitches
+    {
+        public static string enableBinaryFormatterUsageInTypeConverterSerialization = "System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterSerialization";
```

Comment:
We should add a full [feature switch](https://github.com/dotnet/designs/blob/main/accepted/2020/feature-switch.md) for this and it should be defaulted to `false`. So when people trim their apps BinaryFormatter gets trimmed out by default.

See the following for examples on how this is done:

* https://github.com/dotnet/runtime/pull/38963 
* https://github.com/dotnet/runtime/pull/47778

specifically, see the changes to the `ILLink.Substitutions.Shared.xml` file.

Reply:
Nice! I had no idea this existed.

Reply:
Alright, so I 
1. Created an entry in docs/workflow/trimming/feature-switches.md.
2. Added the `get_AllowBinaryFormatterUsage...` method to ILLink.Substitutions.xml file so the method can get stubbed out.
3. Added a test to test/TrimmingTests/. This was primarily to inspect the dll locally to make sure the appropriate parts of the `Serialize` method were being stubbed out. Is there a reason to keep this test?
4. I'm not sure what tests to add to test the linker behavior?

Reply:
>I'm not sure what tests to add to test the linker behavior?

The trimming tests are really just there for correctness testing (making sure the linker won't strip out required code) and not for size testing (making sure you don't increase the app size). Seems like your PR is really focused more on the latter, so perhaps adding a trimming test will only be useful locally for testing yourself that BinaryFormatter-related code is actually stripped, but I wouldn't add a trimming test for it on the PR. The test I would add however, is something that tests the AppContext switch since that one is not really related to the linker, so you can test that as a regular unit test using Remote Executor and setting the switch value and making sure that it works on both BF and non-BF paths.